test/e2e/nsx_subnet_test.go

```diff
@@ -602,3 +699,574 @@ func SubnetValidate(t *testing.T) {
+	createAndWaitForSubnet(t, subnetPathPatch, subnetPathGet, requestBody)
+	log.Info("Using subnet path", "path", subnetPathGet)
+	return subnetPathGet, subnetID
+}
+
+// updateVPCNetworkConfigWithSubnet updates the VPC network configuration for a namespace with a subnet path
+func updateVPCNetworkConfigWithSubnet(t *testing.T, namespace string, subnetPath string) {
+	// Get the vpc_network_config annotation from the namespace
+	nsObj, err := testData.clientset.CoreV1().Namespaces().Get(context.TODO(), namespace, v1.GetOptions{})
+	require.NoError(t, err)
+	vpcNetworkConfigName := nsObj.Annotations[common.AnnotationVPCNetworkConfig]
+	require.NotEmpty(t, vpcNetworkConfigName, "vpc_network_config annotation should not be empty")
+
+	// Get the VPC network configuration using the annotation value
+	vpcNetworkConfig, err := testData.crdClientset.CrdV1alpha1().VPCNetworkConfigurations().Get(context.TODO(), vpcNetworkConfigName, v1.GetOptions{})
+	require.NoError(t, err)
+	vpcNetworkConfig.Spec.Subnets = []string{subnetPath}
+	_, err = testData.crdClientset.CrdV1alpha1().VPCNetworkConfigurations().Update(context.TODO(), vpcNetworkConfig, v1.UpdateOptions{})
+	require.NoError(t, err)
+}
+
+// clearVPCNetworkConfigSubnets removes all subnet paths from the VPC network configuration for a namespace
+func clearVPCNetworkConfigSubnets(t *testing.T, namespace string) {
+	// Get the vpc_network_config annotation from the namespace
+	nsObj, err := testData.clientset.CoreV1().Namespaces().Get(context.TODO(), namespace, v1.GetOptions{})
+	require.NoError(t, err)
+	vpcNetworkConfigName := nsObj.Annotations[common.AnnotationVPCNetworkConfig]
+	require.NotEmpty(t, vpcNetworkConfigName, "vpc_network_config annotation should not be empty")
+
+	// Get the VPC network configuration using the annotation value
+	vpcNetworkConfig, err := testData.crdClientset.CrdV1alpha1().VPCNetworkConfigurations().Get(context.TODO(), vpcNetworkConfigName, v1.GetOptions{})
+	require.NoError(t, err)
+	vpcNetworkConfig.Spec.Subnets = []string{} // Remove all subnet paths
+	_, err = testData.crdClientset.CrdV1alpha1().VPCNetworkConfigurations().Update(context.TODO(), vpcNetworkConfig, v1.UpdateOptions{})
+	require.NoError(t, err)
+}
+
+// listNamespaceSubnets lists all subnets in a namespace
+func listNamespaceSubnets(ctx context.Context, namespace string) (*v1alpha1.SubnetList, error) {
+	subnetList, err := testData.crdClientset.CrdV1alpha1().Subnets(namespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		log.Error(err, "Failed to list subnets", "namespace", namespace)
+	}
+	return subnetList, err
+}
+
+// verifySharedSubnets is a unified function that verifies shared subnets in a namespace
+// Parameters:
+// - t: testing.T instance
+// - namespace: namespace to check
+// - expectedCount: expected number of shared subnets (0 for none, 1 for single, >1 for multiple)
+// - namePrefixes: optional list of name prefixes to validate (ignored if expectedCount is 0)
+// Returns:
```

Comment:
should namePrefixes be a []string, does the length of namePrefixes == expectedCount?

Reply:
namePrefixes is optional list of names, basicly the same with []string